### PR TITLE
SSHDriver cleanup

### PR DESCRIPTION
**Pull request by Emantor**
**Description**
- Remove support for user control master connections
- Document the `cleanup` function for targets in the library case
- Minor adjustments to `argument` printing

**Checklist**
- [x] PR has been tested

### CODE REVIEW

File: doc/usage.rst [hunk 1]
```diff
@@ -177,6 +177,12 @@ The labgrid library provides two ways to configure targets with resources and
 drivers: either create the :any:`Target` directly or use :any:`Environment` to
 load a configuration file.
 
+.. warning::
+   After your script/application is done using labgrid as a library, ensure to
+   call the ``cleanup()`` function all created targets. This will deactivate all
```
Comment (jluebbe) [inline doc/usage.rst:182]: Maybe it would be better to use the [atexit module](https://docs.python.org/3/library/atexit.html) in the Target to trigger this automatically? We'd need to make sure to handle any exceptions gracefully.

File: doc/usage.rst [hunk 2]
```diff
@@ -224,6 +230,12 @@ syntactic sugar::
   >>> target[FakeConsoleDriver, 'console']
   FakeConsoleDriver(target=Target(name='main', …), name='console', …)
 
+After you are done with the target, call the cleanup function to ensure
+drivers are deactivated and processes are stopped:::
+
+  >>> target.clean()
```
Comment (jluebbe) [inline doc/usage.rst:236]: The function name is cleanup(). If we go the atexit approach, this would be optional but could catch some errors earlier.

**Review by jluebbe — CHANGES_REQUESTED** [review]

File: doc/usage.rst (hunk 1 above)
Comment (Emantor) [reply doc/usage.rst:182]: Neat, I'll add that for the target. Handling this should IMO be done by catching the exception, mentioning that an error happened and directing the user to call `cleanup()` himself to get the full exception and investigate the problem.

File: doc/usage.rst (hunk 2 above)
Comment (Emantor) [reply doc/usage.rst:236]: I'll reword this to conform to this use case.

File: doc/usage.rst [hunk 3]
```diff
@@ -224,6 +228,15 @@ syntactic sugar::
   >>> target[FakeConsoleDriver, 'console']
   FakeConsoleDriver(target=Target(name='main', …), name='console', …)
 
+After you are done with the target, optionally call the cleanup function on your
```
Comment (Bastian-Krause) [inline doc/usage.rst:231]:
```suggestion
After you are done with the target, optionally call the cleanup method on your
```

File: doc/usage.rst [hunk 4]
```diff
@@ -224,6 +228,15 @@ syntactic sugar::
...
+target. While labgrid registers an atexit handler to cleanup targets, this has
+the advantage that exceptions can be handled by your application:::
+
+  >>> try:
+  >>>     target.cleanup()
+  >>> except Exception as e:
+  >>> <your code here>
```
Comment (Bastian-Krause) [inline doc/usage.rst:238]:
```suggestion
  >>>     <your code here>
```

File: labgrid/target.py [hunk 5]
```diff
@@ -466,6 +468,14 @@ def deactivate_all_drivers(self):
         for drv in reversed(self.drivers):
             self.deactivate(drv)
 
+    def _atexit_cleanup(self):
+        try:
+            self.cleanup()
+        except Exception as e:
+            print("An exception occured during cleanup, call the cleanup() "
+                  "function on targets yourself to explictly handle exceptions.")
```
Comment (Bastian-Krause) [inline labgrid/target.py:476]:
```suggestion
                  "method on targets yourself to handle exceptions explicitly.")
```

**Review by Bastian-Krause — CHANGES_REQUESTED** [review]

**Review by jluebbe — APPROVED** [review]
